src/LayerEditor/ui/panels/diagram.py

```diff
@@ -106,38 +106,60 @@ def __init__(self, parent=None):
   
         self.set_data()    
         self.setSceneRect(0, 0, 20, 20)
-        
+
+    def remove_graphical_object(self, obj):
+        self.removeItem(obj)
+        self.regionsUpdateRequired.emit()
+
+    def add_graphical_object(self, obj):
+        self.addItem(obj)
+        dim = None
+        layer_id = cfg.diagram.regions.current_layer_id
+        if isinstance(obj, Point):
+            dim = 0
+        elif isinstance(obj, Line):
+            dim = 1
+        elif isinstance(obj, Polygon):
+            dim = 2
+        if dim is not None:
```

Comment:
Add comments.
Seems that this should rather be an assert, and possibly we need check for `obj is None` in beginning of function.